riocli/project/update_owner.py

```diff
@@ -0,0 +1,102 @@
+# limitations under the License.
+import typing
+
+import click
+from click_help_colors import HelpColorsCommand
+from email_validator import EmailNotValidError, validate_email
+
+from riocli.config import get_config_from_context
+from riocli.constants import Colors, Symbols
+from riocli.project.util import name_to_guid
+
+
+@click.command(
+    'update-owner',
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.argument('project-name', type=str, required=True)
+@click.option('--user-email', type=str, help="Email of the new owner")
+@name_to_guid
+@click.pass_context
+def update_owner(
+        ctx: click.Context,
+        project_name: str,
+        project_guid: str,
+        user_email: str,
+) -> None:
+    """
+    Update the owner of the project.
+
+    The command will show an interactive list of users in the project if
+    you do not specify --user-email. You can select the new owner from the list.
+    """
+    config = get_config_from_context(ctx)
+    client = config.new_v2_client(with_project=False)
+
+    try:
+        project = client.get_project(project_guid)
+    except Exception as e:
+        click.secho('{} Failed to fetch project: {}'.format(Symbols.ERROR, e), fg=Colors.RED)
+        raise SystemExit(1)
+
+    project_users = project.get('spec', {}).get('users', [])
+
+    user = None
+
+    if user_email:
+        try:
+            validate_email(user_email)
+        except EmailNotValidError as e:
+            click.secho('{} {} is not a valid email address'.format(Symbols.ERROR, user_email), fg=Colors.RED)
+            raise SystemExit(1) from e
+
+        for u in project_users:
+            if u['emailID'] == user_email:
+                user = u
+                break
+    else:
+        user = show_interactive_selection(project_users, project.get('metadata', {}).get('creatorGUID'))
+
+    if user is None:
+        click.secho('{} User not found in project'.format(Symbols.ERROR), fg=Colors.RED)
+        raise SystemExit(1)
+
+    try:
+        client.update_project_owner(project_guid, user['userGUID'])
+        click.secho('{} Owner updated successfully'.format(Symbols.SUCCESS), fg=Colors.GREEN)
+    except Exception as e:
+        click.secho('{} Failed to update owner: {}'.format(Symbols.ERROR, e), fg=Colors.RED)
+        raise SystemExit(1)
+
+
+def show_interactive_selection(users: typing.List[typing.Dict], current: str) -> typing.Dict:
+    click.secho("Select new project owner:")
+    for i, u in enumerate(users):
+        fmt = '{}. {} {} ({})'.format(
+            i + 1,
+            u.get('firstName', ''),
+            u.get('lastName', ''),
+            u.get('emailID'))
+
+        if u['userGUID'] == current:
+            fmt = click.style(fmt, bold=True, italic=True, fg=Colors.BRIGHT_WHITE)
+
+        click.secho(fmt)
+
+    choice = click.prompt('Choose', type=click.IntRange(1, len(users)))
+
+    return users[choice - 1]
```

Comment:
suggestion: Consider using show_selection function from utils module.

Reply:
It's kind of tricky to use `show_selection` in utils. What I want to show is only the name and the email. But what I want to select is the corresponding GUID which I don't want to show in the list.

The dict selection helper does this. It prints the key as well as the value.
```
fmt = '{}\n{}) {} - {}'.format(fmt, idx + 1, key, ranger[key])
```

I can update this helper just to show the value but I believe this is currently being used at certain places in the CLI and hence did not modify.

Reply:
As discussed, I've updated the helper methods to accommodate this change.